sql/src/main/java/io/crate/analyze/CopyFromAnalyzedStatement.java

```diff
@@ -37,16 +38,24 @@
     @Nullable
     private final String partitionIdent;
     private final Predicate<DiscoveryNode> nodePredicate;
+    private final FileUriCollectPhase.InputFormat inputFormat;
 
     public CopyFromAnalyzedStatement(DocTableInfo table,
                                      Settings settings,
                                      Symbol uri,
                                      @Nullable String partitionIdent,
-                                     Predicate<DiscoveryNode> nodePredicate) {
+                                     Predicate<DiscoveryNode> nodePredicate,
+                                     FileUriCollectPhase.InputFormat inputFormat) {
         super(settings, uri);
         this.table = table;
         this.partitionIdent = partitionIdent;
         this.nodePredicate = nodePredicate;
+        this.inputFormat = inputFormat;
+    }
+
+    @Nullable
```

Comment:
I actually don't think this should be marked as `Nullable ` and have a feeling it is left over from a time before a default value was set.  The `inputFormat()` should never be null, because if there is no input format provided (even if relying on the file extension), then it will have defaulted to `json`.